src/Testcontainers.Xunit/ContainerLifetime.cs

```diff
@@ -0,0 +1,91 @@
+namespace Testcontainers.Xunit;
+
+/// <summary>
+/// Base class managing the lifetime of a container.
+/// </summary>
+/// <typeparam name="TBuilderEntity">The builder entity.</typeparam>
+/// <typeparam name="TContainerEntity">The container entity.</typeparam>
+public abstract class ContainerLifetime<TBuilderEntity, TContainerEntity> : IAsyncLifetime
+    where TBuilderEntity : IContainerBuilder<TBuilderEntity, TContainerEntity>, new()
+    where TContainerEntity : IContainer
+{
+    private readonly Lazy<TContainerEntity> _container;
+
+    [CanBeNull]
+    private ExceptionDispatchInfo _exception;
+
+    protected ContainerLifetime(ILogger logger)
+    {
+        _container = new Lazy<TContainerEntity>(() => Configure(new TBuilderEntity().WithLogger(logger)).Build());
+    }
+
+    /// <summary>
+    /// Gets the container instance.
+    /// </summary>
+    public TContainerEntity Container
+    {
+        get
+        {
+            _exception?.Throw();
+            return _container.Value;
+        }
+    }
+
+    /// <inheritdoc />
+    LifetimeTask IAsyncLifetime.InitializeAsync() => InitializeAsync();
+
+#if !XUNIT_V3
+    /// <inheritdoc />
+    LifetimeTask IAsyncLifetime.DisposeAsync() => DisposeAsync();
+#else
+    /// <inheritdoc />
+    LifetimeTask IAsyncDisposable.DisposeAsync() => DisposeAsync();
+#endif
+
+    /// <summary>
+    /// Extension method to further configure the container instance.
+    /// </summary>
+    /// <example>
+    ///   <code>
+    ///   public class MariaDbRootUserFixture(IMessageSink messageSink) : DbContainerFixture&lt;MariaDbBuilder, MariaDbContainer&gt;(messageSink)
+    ///   {
+    ///     public override DbProviderFactory DbProviderFactory =&gt; MySqlConnectorFactory.Instance;
+    ///     <br />
+    ///     protected override MariaDbBuilder Configure(MariaDbBuilder builder)
+    ///     {
+    ///       return builder.WithUsername("root");
+    ///     }
+    ///   }
+    ///   </code>
+    /// </example>
+    /// <param name="builder">The container builder to configure.</param>
+    /// <returns>A configured instance of <typeparamref name="TBuilderEntity" />.</returns>
+    protected virtual TBuilderEntity Configure(TBuilderEntity builder)
+    {
+        return builder;
+    }
+
+    /// <inheritdoc cref="IAsyncLifetime" />
+    protected virtual async LifetimeTask InitializeAsync()
+    {
+        try
+        {
+            await Container.StartAsync()
+                .ConfigureAwait(false);
+        }
+        catch (Exception e)
+        {
+            _exception = ExceptionDispatchInfo.Capture(e);
```

Comment:
What is the purpose of capturing the exception here? Why don't we fail fast? I mean, as soon as we access the `Container` property, it will throw an exception anyway, e.g. with the `DbContainerFixture` implementation in the `InitializeAsync` method. Do we really need this exception handling?

Dispose is called anyway by the test framework, and Testcontainers should handle cases where the container does not exist as well.

Reply:
Hmm, I don’t really like it. This seems more like something the test framework should handle. If I want to skip tests (in xUnit.net V2) for certain reasons, I would inherit from `FactAttribute` and check it **once** before actually running the test, similar to what we are already [doing](https://github.com/testcontainers/testcontainers-dotnet/blob/4834c3b4eefe1cc3f4f628a7361b94548ea59c51/tests/Testcontainers.Platform.Windows.Tests/WindowsContainerTest.cs#L22).

Documenting this is not easy either and, personally, not what I expect as a library consumer. If the library cannot start a container, I expect an exception immediately. However, if it helps, we can keep it as is for now.